### ISSUE TITLE
'ccache.get_tgs' called with 'strict=False' prevents using ccache with other TGS

### ISSUE THREAD

**Issue by CravateRouge** [skelsec/minikerberos#43]
I understand the `strict=False` mode which can be very convenient in `get_tgs` to use a tgs ticket even if the name is wrong but the ticket is actually valid for the target:
https://github.com/skelsec/minikerberos/blob/45d701fe18fc638d57638086a1e2d15f8abd7083/minikerberos/common/ccache.py#L642

However, it shoudn't be called in core functions without being able to provide a `strict=True` option such as in `tgs_from_ccache` called by `get_TGS`:
https://github.com/skelsec/minikerberos/blob/45d701fe18fc638d57638086a1e2d15f8abd7083/minikerberos/aioclient.py#L390

I think it would be good to provide a `strict` argument in parents functions, or let the user provide a `strict=True` parameter in the connection URL.

What do you think? I can provide the PR.

**Comment by skelsec** [comment]
yeah, that totally makes sense. the `strict` parameter was introduced waaaay back when debugging SMB kerberos auth, but I did not followed up since
